CMakeLists.txt

```diff
@@ -20,16 +20,15 @@ cmake_minimum_required (VERSION 2.6)
 
 project (Proton C)
 
+# Enable C++ now for examples and bindings subdirectories, but make it optional.
+enable_language(CXX OPTIONAL)
+
```

Comment:
You are correct. I think that we could now move the block below somewhere else - it was only here to get the project command in the top level file.